beacon_node/store/src/hot_cold_store.rs

```diff
@@ -3070,7 +3080,12 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
                         "slot" => summary.slot,
                         "reason" => reason,
                     );
-                    state_delete_batch.push(StoreOp::DeleteState(state_root, Some(summary.slot)));
+                    state_delete_batch.push(StoreOp::DeleteState {
+                        state_root,
+                        state_slot: Some(summary.slot),
+                        // Delete the diff as descendants of this state will never be used.
+                        prune_hot_diff: true,
```

Comment:
We need to change this to not delete the diff ancestors of the finalized state (use `closest_layer_slots` somewhere here).

Reply:
We may want to rewrite the `prune_abandoned_forks` code, but if we don't want to do that now, we can take the approach above